### PR TITLE
Iterate all dictionary key types in cast test

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

Increasing test coverage. Per the comment https://github.com/apache/arrow-rs/pull/3482#discussion_r1065506680, the dictionary key types are chose arbitrarily and we can improve it.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow/tests/array_cast.rs [hunk 1]
```diff
@@ -409,19 +409,28 @@ fn get_all_types() -> Vec<DataType> {
...
+
+    let dictionary_key_types =
+        vec![Int8, Int16, Int32, Int64, UInt8, UInt16, UInt32, UInt64];
+
+    let mut dictionary_types = dictionary_key_types
+        .into_iter()
+        .flat_map(|key_type| {
+            vec![
+                Dictionary(Box::new(key_type.clone()), Box::new(Int32)),
+                Dictionary(Box::new(key_type.clone()), Box::new(Utf8)),
```
Comment (tustvold) [inline arrow/tests/array_cast.rs:423]: Could we maybe try LargeUtf8 and LargeBinary as well?

**Review by tustvold — APPROVED** [review]
Reading the comment, I think the issue was the casting of the string offsets, not the dictionary keys. Although more test coverage is always good

**Comment by viirya** [comment]
> What's the rationale for this choice of dictionary keys

This is the comment. I thought it was referring the dictionary keys?

**Comment by tustvold** [comment]
How strange, on my phone it links to https://github.com/apache/arrow-rs/pull/3482#discussion_r1065505943

**Pull request merged**